### PR TITLE
Configurator Loadbalancer V1

### PR DESCRIPTION
**Pull request by pkharat**

### CODE REVIEW

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 1]
```diff
@@ -0,0 +1,655 @@
+import os
+import oslo_messaging as messaging
+
+from gbpservice.nfp.configurator.agents import agent_base
+from oslo_log import log as logging
+from gbpservice.nfp.core.main import Event
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Don't import classes. Import files.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 2]
```diff
@@ -0,0 +1,655 @@
...
+from gbpservice.nfp.configurator.lib import utils
+from gbpservice.nfp.configurator.lib.filter import Filter
+from gbpservice.nfp.core import poll as nfp_poll
+from gbpservice.nfp.configurator.lib import lb_constants
+
+LOG = logging.getLogger(__name__)
+
+
+class LBaasRpcSender(Filter):
+    """Agent side of the Agent to Plugin RPC API."""
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Please follow the class commenting style as done in configurator as that was followed by all.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 3]
```diff
@@ -0,0 +1,655 @@
...
+    #   1.0 Initial version
+    #   2.0 Generic API for agent based drivers
+    #       - get_logical_device() handling changed on plugin side;
+    #       - pool_deployed() and update_status() methods added;
+
+    def __init__(self, sc=None):
+        self.sc = sc
+        pass
+
+    def _notification(self, data):
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Since the notification method is common, we added new class (AgentBaseNotification) in agent_base.py file. Please create an instance of this class and invoke the _notification() method of that class.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 4]
```diff
@@ -0,0 +1,655 @@
...
+        LOG.info("Sending notification: %s" % data)
+        event = self.sc.new_event(
+            id='NOTIFICATION_EVENT', key='NOTIFICATION_EVENT', data=data)
+        self.sc.poll_event(event)
+
+    def get_logical_device(self, pool_id, context=None):
+        # Call goes to filter library
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Function comments please. Follow the format as in configurator.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 5]
```diff
@@ -0,0 +1,655 @@
...
+                'get_logical_device',
+                pool_id=pool_id
+            )
+        )
+
+    def update_status(self, obj_type, obj_id, status):
+        msg = {'receiver': lb_constants.NEUTRON,
+               'resource': lb_constants.SERVICE_TYPE,
+               'method': 'update_status',
+               'kwargs': [{'obj_type': obj_type,
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Only for generic configuration, kwargs is a list. For *aaS notifications, this should be a dict. Please refer firewall agent.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 6]
```diff
@@ -0,0 +1,655 @@
...
+                           'obj_id': obj_id,
+                           'status': status}]
+               }
+        self._notification(msg)
+
+    def update_pool_stats(self, pool_id, stats):
+        LOG.info("[LbaaSRpcSender]: Update pool stats called")
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: We are following a different way of logging for future extendibility. We frame the message and then invoke the logging API. Please check firewall/configurator for reference.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 7]
```diff
@@ -0,0 +1,655 @@
...
+
+"""
+
+
+class LBaaSRpcManager(agent_base.AgentBaseRPCManager):
+    def __init__(self, sc, conf):
+        """Instantiates child and parent class objects.
+
+        Passes the instances of core service controller and oslo configuration
+        to parent instance inorder to provide event enqueue facility for batch
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: There is no batch processing for *aaS. Fix this comment accordingly.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 8]
```diff
@@ -0,0 +1,655 @@
...
+
+        """
+
+        super(LBaaSRpcManager, self).__init__(sc, conf)
+
+    def create_vip(self, context, vip):
+
+        arg_dict = {'context': context,
+                    'vip': vip,
+                    'serialize': True,
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: This will not work. Why are the keys "serialize" and "binding_key" passed inside the data dict? They should be arguments to new_event() method right?

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 9]
```diff
@@ -0,0 +1,655 @@
...
+        self._sc.post_event(ev)
+
+    def delete_vip(self, context, vip):
+
+        arg_dict = {'context': context,
+                    'vip': vip,
+                    'serialize': True,
+                    'binding_key': vip['pool_id']
+                    }
+        ev = self._sc.new_event(id='DELETE_VIP', data=arg_dict)
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: You can have a common method for sending event so that new_event and post_event need not be repeated in all the functions.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 10]
```diff
@@ -0,0 +1,655 @@
...
+"""
+
+
+class LBaaSEventHandler(nfp_poll.PollEventDesc):
+    instance_mapping = {}
+
+    def __init__(self, sc, drivers, rpcmgr, nqueue):
+        self._sc = sc
+        self.drivers = drivers
+        self._rpcmgr = rpcmgr
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Make the naming convention uniform. Remove "_".

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 11]
```diff
@@ -0,0 +1,655 @@
...
+        self.nqueue = nqueue
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: This "nqueue" is not needed anymore. Please remove it. Check generic config agent for example.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 12]
```diff
@@ -0,0 +1,655 @@
...
+        - update pool health monitor
+        - delete pool health monitor
+        - agent updated
+        Enqueues responses into notification queue.
+
+        Returns: None
+
+        """
+
+        # Process single request data blob
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Remove this comment. We only process single data blobs in *aaS agent.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 13]
```diff
@@ -0,0 +1,655 @@
...
+                   "to handle task: %s of topic: %s. "
+                   % (os.getpid(), ev.id, lb_constants.LBAAS_AGENT_RPC_TOPIC))
+            LOG.debug(msg)
+
+            method = getattr(self, "_%s" % (ev.id.lower()))
+            method(ev)
+        except Exception as err:
+            LOG.error("Failed to perform the operation: %s. %s"
+                      % (ev.id, str(err).capitalize()))
+            import traceback
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Need to remove traceback and corresponding prints.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 14]
```diff
@@ -0,0 +1,655 @@
...
+            traceback.print_exc()
+        """
+        finally:
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Is this commented code needed?

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 15]
```diff
@@ -0,0 +1,655 @@
...
+                '''
+                pass
+            else:
+                self._sc.event_done(ev)
+        """
+
+    def _create_vip(self, ev):
+        data = ev.data
+        context = data['context']
+        vip = data['vip']
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Most of the code in these methods is repetitive. Can we have one function for each resource? Like handle_event_vip which covers CRUD of VIP and same with other resources?

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 16]
```diff
@@ -0,0 +1,655 @@
...
+            try:
+                stats = driver.get_stats(pool_id)
+                if stats:
+                    self.plugin_rpc.update_pool_stats(pool_id, stats)
+            except Exception:
+                LOG.exception(_('Error updating statistics on pool %s'),
+                              pool_id)
+
+
+def events_init(sc, drivers, rpcmgr, nqueue):
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Get rid of nqueue object. Not needed now.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 17]
```diff
@@ -0,0 +1,655 @@
...
+    :param rpcmgr: Instance to receive all the RPC messages from configurator
+    module.
+
+    Returns: None
+
+    """
+    evs = [
+        # Events for LBaaS standard RPCs coming from LBaaS Plugin
+        Event(id='CREATE_POOL', handler=LBaaSEventHandler(
+                                                sc, drivers, rpcmgr, nqueue)),
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Better to have a table with all the events and loop to register. Code will be minimal.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 18]
```diff
@@ -0,0 +1,655 @@
...
+
+    """
+    cutils = utils.ConfiguratorUtils()
+    drivers = cutils.load_drivers(lb_constants.DRIVERS_DIR)
+
+    plugin_rpc = LBaasRpcSender(sc)
+
+    for service_type, dobj in drivers.iteritems():
+        '''LB Driver constructor needs plugin_rpc as a param'''
+        instantiated_dobj = dobj(plugin_rpc)
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Does LBaaS driver directly invoke the methods of RPCSender class?

File: gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_lb_driver.py [hunk 19]
```diff
@@ -0,0 +1,792 @@
...
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from neutron import context
+from oslo_log import log as logging
+from gbpservice.nfp.configurator.drivers.loadbalancer.v1.haproxy import haproxy_rest_client
+from gbpservice.nfp.configurator.lib import lb_constants
+from gbpservice.nfp.configurator.drivers.base.base_driver\
```
Comment (njagadish) [inline gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_lb_driver.py]: Fix the multi line separation as per the PEP compliance. Don't use "\"

File: gbpservice/nfp/configurator/lib/lb_constants.py [hunk 20]
```diff
@@ -0,0 +1,50 @@
+# DRIVERS_DIR = '/usr/lib/python2.7/dist-packages/gbpservice/nfp/'\
+#              'configurator/drivers/loadbalancer/'
+DRIVERS_DIR = '/opt/stack/gbp/gbpservice/nfp/configurator/drivers/loadbalancer/'
```
Comment (njagadish) [inline gbpservice/nfp/configurator/lib/lb_constants.py]: We need to fix this driver path specification issue. What if the path changes from setup to setup? How will we take the input.

File: gbpservice/nfp/configurator/test/lb_rpc_sender_v1.py [hunk 21]
```diff
@@ -0,0 +1,620 @@
...
+from oslo_log import log as logging
+from gbpservice.nfp.configurator.agents import loadbalancer_v1
+from neutron.common import rpc as n_rpc
+from neutron import context
+eventlet.monkey_patch()
+
+LOG = logging.getLogger(__name__)
+
+CONFIGURATOR_TOPIC = "configurator"
+HOST = "Devstack-8"
```
Comment (njagadish) [inline gbpservice/nfp/configurator/test/lb_rpc_sender_v1.py]: Why is the hostname hardcoded?

File: gbpservice/nfp/configurator/test/lb_rpc_sender_v1.py [hunk 22]
```diff
@@ -0,0 +1,620 @@
...
+from oslo_config import cfg
+from oslo_log import log as logging
+from gbpservice.nfp.configurator.agents import loadbalancer_v1
+from neutron.common import rpc as n_rpc
+from neutron import context
+eventlet.monkey_patch()
+
+LOG = logging.getLogger(__name__)
+
+CONFIGURATOR_TOPIC = "configurator"
```
Comment (njagadish) [inline gbpservice/nfp/configurator/test/lb_rpc_sender_v1.py]: Why is configurator topic needed in LB unit test cases?

File: gbpservice/nfp/configurator/test/lb_rpc_sender_v1.py [hunk 23]
```diff
@@ -0,0 +1,620 @@
...
+        print 'called test_create_pool'
+        pool_info = {'context': {'service_info':
+                                 self.lb_context.get_service_info()},
+                     'pool': self.lb_resources.pool,
+                     'driver_name': 'loadbalancer'
+                     }
+
+        pool_data = self._prepare_request_data('pool', pool_info)
+        cctxt = self.client.prepare(server=self.host)
+        return cctxt.call(self.context, 'create_network_service_config',
```
Comment (njagadish) [inline gbpservice/nfp/configurator/test/lb_rpc_sender_v1.py]: Are we really making the RPC calls? We need to mock them right? Also, we need not send RPC calls to configurator. Since this is unit testing of LB, we need to send RPCs to LB agent and test.

File: gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py [hunk 24]
```diff
@@ -0,0 +1,346 @@
+import unittest
+import mock
+
```
Comment (njagadish) [inline gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py]: For this file, apart from the comments provided here, please refer the review comments provided for firewall as well.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py (hunk 1 above)
Comment (pkharat) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Addressed

File: gbpservice/nfp/configurator/test/lb_rpc_sender_v1.py (hunk 22 above)
Comment (pkharat) [inline gbpservice/nfp/configurator/test/lb_rpc_sender_v1.py]: This is a test file for informal unit testing

File: gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py [hunk 25]
```diff
@@ -4,537 +4,197 @@
 from oslo_log import log as logging
 
 
-from gbpservice.nfp.modules import configurator as cfgr
+from gbpservice.nfp.modules import configurator
 from gbpservice.nfp.configurator.agents import loadbalancer_v1 as lb
-from gbpservice.nfp.configurator.agents import generic_config as gc
 from gbpservice.nfp.configurator.drivers.loadbalancer.v1.haproxy.haproxy_lb_driver import HaproxyOnVmDriver
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py]: We should import individual classes. Please import file.

File: gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py [hunk 26]
```diff
@@ -4,537 +4,197 @@
...
-from gbpservice.nfp.configurator.lib import demuxer as demuxer_lib
+from gbpservice.nfp.configurator.lib import demuxer
+
+from test_input_data import FakeObjects
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py]: Import file and not individual classes

File: gbpservice/nfp/test/unit/configurator/agent/test_input_data.py [hunk 27]
```diff
@@ -0,0 +1,246 @@
...
+              'UPDATE_VIP': 'update_network_device_config',
+              'CREATE_POOL': 'create_network_device_config',
+              'DELETE_POOL': 'delete_network_device_config',
+              'UPDATE_POOL': 'update_network_device_config',
+              'CREATE_MEMBER': 'create_network_device_config',
+              'DELETE_MEMBER': 'delete_network_device_config',
+              'UPDATE_MEMBER': 'update_network_device_config',
+              'CREATE_HEALTH_MONITOR': 'create_network_device_config',
+              'DELETE_HEALTH_MONITOR': 'delete_network_device_config',
+              'UPDATE_HEALTH_MONITOR': 'update_network_device_config'}
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/test_input_data.py]: Should add test case for COLLECT_STATS  method also

File: gbpservice/nfp/test/unit/configurator/agent/test_input_data.py [hunk 28]
```diff
@@ -0,0 +1,246 @@
...
+
+    def fake_request_data_vip(self):
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/test_input_data.py]: Please follow one pattern for naming like for vip - fake_request_data_vip() is used for both create and update but for member fake_request_data_create_member() used for both create and delete. We can do this way, just have single method per resource which has common data needed for  create,update,delete. e.g fake_request_data_vip(). Based on the operation add required data by appending the key-value in the request_data['config']['kwargs']

File: gbpservice/nfp/test/unit/configurator/agent/test_input_data.py [hunk 29]
```diff
@@ -0,0 +1,246 @@
...
+            "weight": 1,
+            "address": "42.0.0.11",
+            "tenant_id": "f6b09b7a590642d8ac6de73df0ab0686",
+            "protocol_port": 80,
+            "id": "4910851f-4af7-4592-ad04-08b508c6fa21",
+            "pool_id": "6350c0fd-07f8-46ff-b797-62acd23760de"}]
+        return member
+
+    def _fake_hm_obj(self):
+        hm = [{
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/test_input_data.py]: Why do we need member and hm as list of dictionaries ? It can be taken as dictionary only.

File: gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py [hunk 30]
```diff
@@ -4,537 +4,197 @@
...
 
     def __init__(self, *args, **kwargs):
-        super(ControllerTestCase, self).__init__(*args, **kwargs)
-        self.lb = FakeObjects()
+        super(LBaaSRpcManagerTest, self).__init__(*args, **kwargs)
+        self.fo = FakeObjects()
+        self.arg_dict_vip = {
+            'context': self.fo.context,
+            'vip': self.fo._fake_vip_obj(),
+            'serialize': True,
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py]: Need to correct this serialize and binding_key args. This should  not be set in arg_dict but should be set in the event object. Please check the latest lb agent code

File: gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py [hunk 31]
```diff
@@ -4,537 +4,197 @@
...
+        agent = lb.LBaaSRpcManager(sc, conf)
         return agent, sc
 
-    @mock.patch(__name__ + '.FakeObjects.nqueue')
-    @mock.patch(__name__ + '.FakeObjects.drivers')
-    def _get_GenericConfigEventHandler_object(
-            self, sc, rpcmgr, drivers, nqueue):
-        agent = gc.GenericConfigEventHandler(sc, drivers, rpcmgr, nqueue)
-        return agent
+    def _test_controller(self, operation):
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py]: Please change function name to test_event_creation() as we are creating events. _test_controller() does not convey much info here.

File: gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py [hunk 32]
```diff
@@ -4,537 +4,197 @@
...
                               return_value=agent):
-            if method[operation] == 'create_network_device_config' or method[operation]=='delete_network_device_config':
-                request_data = self.lb.fake_request_data_vip()
-                args = arg_dict
-            elif method[operation] == 'update_network_device_config':
-                request_data = self.lb.fake_request_data_vip_update()
-                args = arg_dict_update
-                
-            getattr(rpc_mgr, method[operation])(self.lb.context, request_data)
+            if operation == 'CREATE_VIP' or operation == 'DELETE_VIP':
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py]: We don't need these if else blocks if we pass (operation,request_data,args) to _test_controller() from the respective functions.Please change accordingly

File: gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py [hunk 33]
```diff
@@ -4,537 +4,197 @@
...
+
+    @mock.patch(__name__ + '.FakeObjects.conf')
+    @mock.patch(__name__ + '.FakeObjects.sc')
+    def _get_configurator_rpc_manager_object(self, sc, conf):
+        cm = configurator.ConfiguratorModule(sc)
+        dmx = demuxer.ConfiguratorDemuxer()
+        rpc_mgr = configurator.ConfiguratorRpcManager(sc, cm, conf, dmx)
+        return sc, conf, rpc_mgr
+
+    def test_notifications(self):
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py]: Don't we need to add test cases for update_status() update_pool_stats(), get_logical_device() ?

File: gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py [hunk 34]
```diff
@@ -4,537 +4,197 @@
...
-            'driver_name': 'haproxy',
-            'host': fo.host,
-            'kwargs': kwargs,
-        }
-        self.id = 'dummy'
+        self._test_controller('UPDATE_HEALTH_MONITOR')
+        pass
 
 
 class LBaasHandlerTestCase(unittest.TestCase):
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/lb_test_agent.py]: Please rename class name to LBaasEventHandlerTestCase

File: gbpservice/nfp/test/unit/configurator/agent/lb_test_driver.py [hunk 35]
```diff
@@ -1,360 +1,18 @@
-import requests
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/lb_test_driver.py]: Please give other file name. lb_test_driver.py is generic one but thi is HaproxyOnVmDriver.

File: gbpservice/nfp/test/unit/configurator/agent/lb_test_driver.py [hunk 36]
```diff
@@ -1,360 +1,18 @@
...
-            'old_health_monitor': 'oldhm',
-            'pool_id': '6350c0fd-07f8-46ff-b797-62acd23760de',
-            'driver_name': 'haproxy',
-            'host': fo.host,
-            'kwargs': kwargs,
-        }
-        self.id = 'dummy'
-
-
 class LbaasDriverTestCase(unittest.TestCase):
```
Comment (pkharat) [inline gbpservice/nfp/test/unit/configurator/agent/lb_test_driver.py]: Plz change class name to HaproxyOnVmDriverTestCase

File: gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_rest_client.py [hunk 37]
```diff
@@ -0,0 +1,152 @@
...
+            msg = ("Request failed in REST Api Server. %s"
+                   % str(err).capitalize())
+            LOG.error(msg)
+
+        if response is None:
+            # Request was timed out.
+            msg = ("Response is Null, Request for method: %s to "
+                   "URI: %s timed out" % (method, uri))
+            LOG.error(msg)
+            # TODO(Magesh): Use constants defined in requests or httplib
```
Comment (njagadish) [inline gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_rest_client.py]: Can you use constants as per the comment here?

File: gbpservice/nfp/configurator/lib/lb_constants.py [hunk 38]
```diff
@@ -0,0 +1,69 @@
+DRIVERS_DIR = '/usr/lib/python2.7/dist-packages/gbpservice/nfp/'\
```
Comment (njagadish) [inline gbpservice/nfp/configurator/lib/lb_constants.py]: License header needs to be added.

Comment (njagadish) [inline gbpservice/nfp/configurator/lib/lb_constants.py]: Avoid "\" separator

File: gbpservice/nfp/configurator/test/lb_rpc_sender_v1.py [hunk 39]
```diff
@@ -0,0 +1,626 @@
+import eventlet
```
Comment (njagadish) [inline gbpservice/nfp/configurator/test/lb_rpc_sender_v1.py]: Add License header

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 40]
```diff
@@ -0,0 +1,729 @@
+import os
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Add license header

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py [hunk 41]
```diff
@@ -0,0 +1,729 @@
...
+        self._handle_event_pool_health_monitor(ev, 'create')
+
+    def _update_pool_health_monitor(self, ev):
+        self._handle_event_pool_health_monitor(ev, 'update')
+
+    def _delete_pool_health_monitor(self, ev):
+        self._handle_event_pool_health_monitor(ev, 'delete')
+
+    def _agent_updated(self, ev):
+        """ TODO:(pritam): Support """
```
Comment (njagadish) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: Do we need to address this TODO?

File: gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_lb_driver.py [hunk 42]
```diff
@@ -0,0 +1,791 @@
+# Copyright 2013 New Dream Network, LLC (DreamHost)
```
Comment (njagadish) [inline gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_lb_driver.py]: This looks like a different license. Please use the same license that was used in other files.

File: gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_lb_driver.py [hunk 43]
```diff
@@ -0,0 +1,791 @@
...
+        self.plugin_rpc = plugin_rpc
+
+    def _get_rest_client(self, ip_addr):
+        client = haproxy_rest_client.HttpRequests(
+                            ip_addr, lb_constants.HAPROXY_AGENT_LISTEN_PORT,
+                            REQUEST_RETRIES, REQUEST_TIMEOUT)
+        return client
+
+    def _get_device_for_pool(self, pool_id, context):
+        device = HaproxyOnVmDriver.pool_to_device.get(pool_id, None)
```
Comment (njagadish) [inline gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_lb_driver.py]: All functions should have function headers. We can do this in a different PR though.

File: gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_rest_client.py [hunk 44]
```diff
@@ -0,0 +1,152 @@
+# One Convergence, Inc. CONFIDENTIAL
+# Copyright (c) 2012-2015, One Convergence, Inc., USA
```
Comment (njagadish) [inline gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_rest_client.py]: Use license as in other files.

File: gbpservice/nfp/configurator/agents/loadbalancer_v1.py (hunk 41 above)
Comment (pkharat) [inline gbpservice/nfp/configurator/agents/loadbalancer_v1.py]: We need to address this but as next step.

**Comment by njagadish** [comment]
Merging this PR. Further changes/fixes will be tracked with different PRs.
